pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix

```
@@ -106,6 +106,31 @@ self: let
         };
       });
 
+      jinx = super.jinx.overrideAttrs (old: {
+        dontUnpack = false;
+
+        nativeBuildInputs = (old.nativeBuildInputs or [ ]) ++ [
+            pkgs.pkg-config
+        ];
+
+        buildInputs = (old.buildInputs or [ ]) ++ [ pkgs.enchant2 ];
+
+        postBuild = ''
+          NIX_CFLAGS_COMPILE="$($PKG_CONFIG --cflags enchant-2) $NIX_CFLAGS_COMPILE"
+          $CC -shared -o jinx-mod.so jinx-mod.c -lenchant-2
+        '';
+
+        postInstall = (old.postInstall or "") + "\n" + ''
+          outd=$(echo $out/share/emacs/site-lisp/elpa/jinx-**)
+          install -m444 --target-directory=$outd jinx-mod.so
+          rm $outd/jinx-mod.c $outd/emacs-module.h
```

```suggestion
          outd=$out/share/emacs/site-lisp/elpa/jinx-*
          install -m444 -t $outd jinx-mod.so
          rm $outd/jinx-mod.c $outd/emacs-module.h
```

If you want to match files recursively by using `**`, it is not needed.  And I am not sure if shell option `globstar` is enabled.